src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,82 @@
-import React from 'react';
+import React, {
+  FormEvent, useCallback, useMemo, useState,
+} from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+import { ResponseError } from '../../types/ReponseError';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+
+interface Props {
+  addToList: (movie: Movie) => void,
+}
+
+export const FindMovie: React.FC<Props> = ({ addToList }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [error, setError] = useState<ResponseError | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = useCallback((event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  }, []);
+
+  const handleSearchSubmit = useCallback(async () => {
+    try {
+      setLoading(true);
+
+      const movieData = await getMovie(query);
+
+      if ('Error' in movieData) {
+        setError(movieData);
+        setMovie(null);
+      } else {
+        const {
+          Title,
+          Plot,
+          Poster,
+          imdbID,
+        } = movieData;
+
+        setMovie({
+          title: Title,
+          description: Plot,
+          imgUrl: Poster,
+          imdbId: imdbID,
+          imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+        });
+
+        setError(null);
+      }
+    } catch {
+      setError({ Response: 'False', Error: 'unexpected error' });
+      setMovie(null);
+    } finally {
+      setLoading(false);
+    }
+  }, [query]);
+
+  const handleQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+      setError(null);
+    }, [],
+  );
+
+  const buttonDisabled = useMemo(() => query === '', [query]);
```

Comment:
redundant to cover all vars or functions in useMemo or useCallback

use these hooks only when necessary

Reply:
1. Expensive computations: If you have a function that performs heavy calculations or data processing, you can use useMemo to memoize the result and avoid recomputing it on every render.
2. Rendering optimization: If a component depends on a value that doesn't change often, you can use useMemo to memoize the derived value and prevent unnecessary re-renders.